### ISSUE TITLE
Upgrade to 0.13.2 breaks init behavior, rejects all backend-config variables other than bucket and dynamo_table

### ISSUE THREAD

**Issue by bedge** [hashicorp/terraform#26138]
### Terraform Version

```
Terraform v0.13.2
```

### Terraform Configuration Files

```terraform
...
```

### Debug Output

```
terraform init --backend-config=../../backends//variables.tfvars

Initializing the backend...

Error: Unsupported argument

  on ../../backends//variables.tfvars line 4:
   4: account_name = ""

An argument named "account_name" is not expected here.


Error: Unsupported argument

  on ../../backends//variables.tfvars line 6:
   6: ami_branch = "develop"

An argument named "ami_branch" is not expected here.


Error: Unsupported argument

  on ../../backends//variables.tfvars line 12:
  12: db_instance_class = "db.r4.xlarge"

An argument named "db_instance_class" is not expected here.


Error: Unsupported argument

  on ../../backends//variables.tfvars line 14:
  14: environment = ""

An argument named "environment" is not expected here.


Error: Unsupported argument

  on ../../backends//variables.tfvars line 16:
  16: release_cycle = "5.1.0"

An argument named "release_cycle" is not expected here.


Error: Unsupported argument

  on ../../backends//variables.tfvars line 18:
  18: org_env = "org-"

An argument named "org_env" is not expected here.


Error: Unsupported argument

  on ../../backends//variables.tfvars line 20:
  20: org_name = "org"

An argument named "org_name" is not expected here.


```

Where ../../backends//variables.tfvars contains all our env-specific variables, including region, bucket and dynamo_table.

The only variables it doesn't complain about are region, bucket and dynamo_table.

Is it now illegal to use this file for other variables too?

### Expected Behavior

I expected terraform to init the same way it always has, ignoring variables it doesn't need for it's own init.

This allows up to use the same file for all env-specific config, rather than requiring a separate file for --backend-config and  -file.

### Actual Behavior

See above, complaints about every variable that was not region, bucket or dynamo_table.

### Steps to Reproduce

```
terraform init --backend-config variables.tfvars 
```
Where variables.tfvars contains:
```
account_name = "xx"

ami_branch = "develop"

bucket = "org-envxx-tfstate"

dynamodb_table = "tfstate-lock-envxx"

db_instance_class = "db.r4.xlarge"

environment = "envxx"

release_cycle = "5.1.0"

org_env = "org-envxx"

org_name = "org"

profile = ""

region = "eu-central-1"
```

**Comment by alisdair** [comment]
Thanks for the detailed report!

> Is it now illegal to use this file for other variables too?

Yes, and it always has been _intended_ to be the case. The fact that Terraform prior 0.13 ignored unexpected attributes was unintentional. [I wrote up some more detail in this comment](https://github.com/hashicorp/terraform/issues/26038#issuecomment-683000810) which explains how we arrived at this more strict interpretation of the backend config file, after multiple bug reports.

I'm sorry that this is impacting your workflow, but hopefully it's clear how to update your configuration: move all backend-specific configuration overrides to a separate file.
